ugs-core/src/com/willwinder/universalgcodesender/AbstractController.java

```diff
@@ -234,8 +234,8 @@ protected void softReset() throws Exception {
     }
 
     @Override
-    public void jogMachine(int dirX, int dirY, int dirZ, double stepSize,
-            double feedRate, UnitUtils.Units units) throws Exception {
+    public void jogMachine(float dirX, float dirY, float dirZ, double stepSize,
+                           double feedRate, UnitUtils.Units units) throws Exception {
```

Comment:
This is kind of cool, but having the direction + step size combination is a little confusing. It looks like you got it right when generating gcode (multiplying the two together).

Is this function still easy to work with? Would we lose anything by changing the method to `jogMachine(float distanceX, float distanceY, float distanceZ, double feedRate, UnitUtils.Units units)`?

Maybe from the joysticks perspective it's natural to have the direction be a ratio.

Reply:
I had to make a choice about a year ago to either create a completely new function for jogging all axises or to copy the direction concept when implementing the jog controller to keep things consistent.

A change like this will ripple all the way up to the jog service, but in the end it would probably become a cleaner solution. I can give it a try!

Reply:
It's up to you, this would definitely cause a big ripple.

Reply:
haha, too late on this comment. I think this will be nice! The function is definitely more intuitive this way.

Reply:
Things got a lot simpler using distances instead of directional flags. I prefer things this way. =)